contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/common.py

```diff
@@ -273,7 +274,7 @@ def __init__(self, code, severity, filename, message, line_range=None, lines=Non
   def __str__(self):
     """convert ascii for safe terminal output"""
     flat = list(self.flatten_lines([self.message], self.lines))
-    return '\n     |'.join(flat).encode('ascii', errors='replace')
+    return '\n     |'.join(flat).encode('ascii', errors='replace').decode('ascii')
```

Comment:
`__str__` must return unicode in Py3 no matter what. 

This function will  still strip out any non-ascii, though, and then spit it back out as unicode.

Reply:
That's probably comment worthy.